### ISSUE TITLE
Using mongodb-core with lodash

### ISSUE THREAD

**Issue by rgmembreno** [mongodb-js/mongodb-core#290]
Hi all,

While trying out the MongoDB native driver for node.js we received an interesting error when trying to connect to our Mongo instance hosted on mLab:

>  MongoError: failed to connect to server [<instance>.mlab.com:<port>] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]

What was really weird about this error is it did not occur when connecting via Mongoose using the same uri and options.  We finally tracked down the root cause today: in mongodb-core/lib/utils.js the assign() function toggles between the native Object.assign() function and the default method used for pre-ES6 versions of node.  We are also using lodash, and that package was replacing the native implementation of Object.assign with that package's version of the method.  The end result is the properties sent to the Server's connection Pool were not getting merged correctly, and it tried to connect to the Mongo instance running on "localhost:27017" instead,

The band-aid I applied was to use the pre-ES6 function exclusively and never use Object,assign.  This problem does not occur in Mongoose because it does not use lodash internally, so the behavior of Object,assign() is not affected.  Has anyone encountered a similar issue when working with lodash and mongodb-core as well in node version  4.0+?  Is there a recommended workaround in this situation that we should use instead of making a new util method that always uses the built-in assign()?

Note that I tested with the following versions of each package and the problem persisted as long as lodash is installed:
* mongodb: 2.2.31 (mongodb-core 2.1.15), 3.0.4+ (mongodb-core 3.0.4+)
* lodash: 3.10.1, 4.11.14

**Comment by mbroadst** [comment]
Hi @rgmembreno, if that's really what's going on that sounds like _very_ serious bug in lodash (though this seems unlikely). In the 3.x series of the driver we have removed the custom `assign` implementation, in favor of _always_ using `Object.assign`, and are unlikely to go back. Are you sure something else isn't polyfilling `Object.assign`? Can you perhaps try to use just the pieces of lodash that you need, rather than bringing the entire module?

**Comment by rgmembreno** [comment]
Hi @mbroadst thanks for the quick reply!

It would be a big change to import only a handful of packages from lodash, and we don't know offhand how long it will take.  It also looks like the problem in lodash is they assume that every core function is available with prototype.<method>.call(), but assign() does not appear on the prototype as of node v 6.12.x  We're sure it's lodash because every time we hit an "Object.assign" in mongodb-core it opens up lodash while debugging.

So we should fork mongodb-core and restore the old behavior?  In the meantime we can try moving off of lodash modules that mutate object properties, but as mentioned above it requires a long time to double check our work.

**Comment by rgmembreno** [comment]
Hi @mbroadst 

Great news, thanks to your pointers we were able to determine that lodash was *not* polyfilling the Object.assign() method automatically,  It took a lot of effort but we finally found the other 3rd party module that was doing this polyfill and updated it to use the native node v6 version of the function, and now everything is working great.

Thank you for all your help with this issue--we never could have figured out the polyfill problem until you asked us to check out the lodash source code to file the bug!  We're all good here, and thanks for answering all of our questions!
